### DISCUSSION TITLE
Ordering of components in a Flex affects zIndex and scaling?

### DISCUSSION THREAD

**Discussion by adamhp** [chakra-ui/chakra-ui#3219] [Q&A]
Hi all,

I have a Flex component with a few items (cards, basically) and I'm using [vanilla-tilt](https://micku7zu.github.io/vanilla-tilt.js/) to give the cards a slight tilt, and also to scale them on hover.

There is this odd behavior where hovering over a card at first behaves correctly, but once a card that comes later in the Flex ordering has been hovered, the previous cards all appear under the subsequent cards (on the z-axis).

I was just hoping that someone might have some idea on why this behavior is occurring. Hopefully this .gif illustrates a bit better. You'll notice the first time I mouse over the first card, it overlaps correctly. The second card also overlaps the first correctly (because it comes later in the Flex order, I presume). The third time I hover over the first card, it appears under the second. The desired behavior is that it overlaps the second again.

![Kapture 2021-01-30 at 12 21 12](https://user-images.githubusercontent.com/7103990/106363306-acc47d80-62f5-11eb-90f1-1ed8eba458d9.gif)

The general structure is (the Card is controlling the z-Index and hover state):
```
<Wrap>
     <WrapItem>
          <Card />
     </WrapItem>
</Wrap>
```

I tried moving the control of state to the `<WrapItem>`, to the `<Tilt>` component within the Card. I also tried explicitly setting the default zIndex of every other component to 0, and also tried setting the zIndex of the cards a bit higher. I've been trying to use Chrome Dev tools to inspect the Elements and check styling to see what I'm missing but I can't quite figure out what is going on. I can't tell if this is a `vanilla-tilt` issue or a Chakra issue, or a PEBCAK... 

Any ideas?

**Comment by chasinhues** [comment]
Could you try testing what happens if you are just using plain CSS (no Chakra)? I mean set it up with the current flex behaviour you've implemented with the ordering etc. as well as the z-index stuff. If you can still observe similar issue then, at least you can rule Chakra out and in that case I'd imagine it might be an issue with the tilt package?

**Reply by adamhp** [reply]
Definitely. Honestly the tilt package was my first guess at a culprit anyway. I was just hoping someone might have seen similar behavior. I'll try isolating it. 

Thanks :)
